test/NuGet.Versioning.Test/VersionRangeFloatParsingTests.cs

```diff
@@ -0,0 +1,159 @@
+
+            List<NuGetVersion> versions = new List<NuGetVersion>()
+            {
+                NuGetVersion.Parse("1.0.0"),
+                NuGetVersion.Parse("1.1.0")
+            };
+
+            Assert.Equal("1.0.0", range.FindBestMatch(versions).ToNormalizedString());
+        }
+
+        [Fact]
+        public void VersionRangeFloatParsing_FloatPrerelease()
+        {
+            var range = VersionRange.Parse("1.0.0-*");
+
+            List<NuGetVersion> versions = new List<NuGetVersion>()
+            {
+                NuGetVersion.Parse("1.0.0-alpha"),
+                NuGetVersion.Parse("1.0.0-beta")
+            };
+
+            Assert.Equal("1.0.0-beta", range.FindBestMatch(versions).ToNormalizedString());
+        }
+
+        [Fact]
+        public void VersionRangeFloatParsing_FloatPrereleaseMatchVersion()
+        {
+            var range = VersionRange.Parse("1.0.0-*");
+
+            List<NuGetVersion> versions = new List<NuGetVersion>()
+            {
+                NuGetVersion.Parse("1.0.0-beta"),
+                NuGetVersion.Parse("1.0.1-omega"),
+            };
+
+            Assert.Equal("1.0.0-beta", range.FindBestMatch(versions).ToNormalizedString());
+        }
+
+        [Fact]
+        public void VersionRangeFloatParsing_FloatPrereleasePrefix()
+        {
+            var range = VersionRange.Parse("1.0.0-beta.*");
+
+            List<NuGetVersion> versions = new List<NuGetVersion>()
+            {
+                NuGetVersion.Parse("1.0.0-beta.1"),
+                NuGetVersion.Parse("1.0.0-beta.2"),
+                NuGetVersion.Parse("1.0.0-omega.3"),
+            };
+
+            Assert.Equal("1.0.0-beta.2", range.FindBestMatch(versions).ToNormalizedString());
+        }
+
+        [Fact]
+        public void VersionRangeFloatParsing_FloatPrereleasePrefixSemVerLabelMix()
+        {
+            var range = VersionRange.Parse("1.0.0-beta.*");
+
+            List<NuGetVersion> versions = new List<NuGetVersion>()
+            {
+                NuGetVersion.Parse("1.0.0-beta.1"),
+                NuGetVersion.Parse("1.0.0-beta.2"),
+                NuGetVersion.Parse("1.0.0-beta.a"),
+            };
+
+            Assert.Equal("1.0.0-beta.a", range.FindBestMatch(versions).ToNormalizedString());
+        }
+
+        [Theory]
+        [InlineData("[]")]
+        [InlineData("[*]")]
+        [InlineData("[1.0.0, 1.1.*)")]
+        [InlineData("[1.0.0, 2.0.*)")]
+        [InlineData("(, 2.*.*)")]
+        [InlineData("<1.0.*")]
+        [InlineData("<=1.0.*")]
+        [InlineData("1.0.0<")]
+        [InlineData("1.0.0~")]
+        [InlineData("~1.*.*")]
+        [InlineData("~*")]
+        [InlineData("~")]
+        [InlineData("^")]
+        [InlineData("^*")]
+        [InlineData(">=*")]
+        [InlineData("1.*.0")]
+        [InlineData("1.*.0-beta-*")]
+        [InlineData("1.*.0-beta")]
+        [InlineData("1.0.0.0.*")]
+        //[InlineData("1.0.0*")]
```

Comment:
There were additional tests for more range operators originally, but I left out implementing those for now. I kept the invalid checks in since either way they are still invalid.